src/utils.js

```diff
@@ -760,8 +760,28 @@ function withExactlyOneSubschema(
   );
 }
 
-function mergeSchemas(schema1, schema2) {
-  return mergeObjects(schema1, schema2, true);
+// Recursively merge deeply nested schemas.
+// The difference between mergeSchemas and mergeObjects
+// is that mergeSchemas only concats arrays for
+// values under the "required" keyword.
+export function mergeSchemas(obj1, obj2) {
+  var acc = Object.assign({}, obj1); // Prevent mutation of source object.
+  return Object.keys(obj2).reduce((acc, key) => {
+    const left = obj1 ? obj1[key] : {},
+      right = obj2[key];
+    if (obj1 && obj1.hasOwnProperty(key) && isObject(right)) {
+      acc[key] = mergeSchemas(left, right);
+    } else if (
+      key === "required" &&
```

Comment:
Can you add some kind of check for the parent property ?

I mean techically speaking, only `properties.required` field should be overwritten, not all fields with `required` (I know it is most unlikely, so if you think this test is enough that would be ok for me)

Reply:
I've added a check to make sure that `obj1` or `obj2` are objects. Not sure if that's the right logic, though, feedback welcome!

Reply:
I've tried with this on your PR, and it still works : 

```
obj1 && obj2 && getSchemaType(obj1) === "object" &&
        getSchemaType(obj2) === "object"
```

I think this is more correct to check for both the schematypes of both objects.

Reply:
you've written `getSchematype(obj1) === "object" || getSchemaType(obj2) === "object"`.

Is the `or` on purpose ? 

To me it seems that it should be an `and` (`&&`), that way only two objects schemas of type object which have a required property will be merged that way ?

Reply:
I was thinking about the case in which you want to merge, say, `{properties: {...}, required: ["a"]}` and `{title: "a"}`. In this case, the second schema is not unambiguously an object, although the first schema is. It would still be nice to merge the two schemas together, though. This would only work if we use `||` instead of `&&`.

Reply:
Ok